plugins/linear-genome-view/src/index.ts

```diff
@@ -142,6 +146,51 @@ export default class LinearGenomeViewPlugin extends Plugin {
         },
       })
     }
+
+    this.initFromLocString(pluginManager)
+  }
+
+  async initFromLocString(pluginManager: PluginManager) {
+    const parsed = queryString.parse(window.location.search)
+    if (pluginManager.rootModel && parsed.loc) {
+      if (!pluginManager.rootModel.session) {
+        // @ts-ignore types are not accurate here, they are stub abstracts
+        pluginManager.rootModel.setSession({ name: 'New session' })
+      }
+      const { session } = pluginManager.rootModel
+
+      // @ts-ignore types are not accurate here, they are stub abstracts
+      const { assemblyManager } = session
+      // @ts-ignore types are not accurate here, they are stub abstracts
+      const view = session.addView('LinearGenomeView', {
+        id: shortid(),
+        type: 'LinearGenomeView',
+      })
+      await when(() => {
+        return (
+          assemblyManager.allPossibleRefNames &&
+          assemblyManager.allPossibleRefNames.length
+        )
+      })
+      const assembly = assemblyManager.assemblies[0]
```

Comment:
probably need to also support assembly name in the locstring

Reply:
Otherwise we could manually parse the locstring ourselves and avoid relying on the complex behavior of navToLocString...